mindsdb/integrations/handlers/elasticsearch_handler/elasticsearch_handler.py

```diff
@@ -0,0 +1,240 @@
+from typing import Optional
+from collections import OrderedDict
+
+import pandas as pd
+from elasticsearch import Elasticsearch
+
+from mindsdb_sql import parse_sql
+from mindsdb_sql.render.sqlalchemy_render import SqlalchemyRender
+from es.elastic.sqlalchemy import ESDialect
+from mindsdb.integrations.libs.base_handler import DatabaseHandler
+
+from mindsdb_sql.parser.ast.base import ASTNode
+
+from mindsdb.utilities.log import log
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+
+class ElasticsearchHandler(DatabaseHandler):
+    """
+    This handler handles connection and execution of the Airtable statements.
+    """
+
+    name = 'elasticsearch'
+
+    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+        """
+        Initialize the handler.
+        Args:
+            name (str): name of particular handler instance
+            connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'elasticsearch'
+
+        if ('hosts' not in connection_data) and ('cloud_id' not in connection_data):
+            raise Exception("Either the hosts or cloud_id parameter should be provided!")
+
+        optional_parameters = ['hosts', 'cloud_id', 'username', 'password']
+        for parameter in optional_parameters:
+            if parameter not in connection_data:
+                connection_data[parameter] = None
+
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> StatusResponse:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        if self.is_connected is True:
+            return self.connection
+
+        self.connection = Elasticsearch(
+            hosts=self.connection_data['hosts'].split(','),
+            cloud_id=self.connection_data['cloud_id'],
```

Comment:
Should we add cloud_id, user, pass to the README file?

Reply:
@ZoranPandovski I have already added it here,
![image](https://user-images.githubusercontent.com/49385643/185951386-ad798bbe-5f0e-49b0-b28b-192e92f32be2.png)

Is this what you meant?